plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.ts

```diff
@@ -56,14 +63,23 @@ export class BitbucketServerEntityProvider implements EntityProvider {
   private readonly logger: LoggerService;
   private readonly scheduleFn: () => Promise<void>;
   private connection?: EntityProviderConnection;
+  private readonly catalogApi?: CatalogApi;
+  private readonly events?: EventsService;
+  private readonly tokenManager?: TokenManager;
+  private eventConfigErrorThrown = false;
+  private readonly targetAnnotation: string;
+  private readonly defaultBranchAnnotation: string;
 
   static fromConfig(
     config: Config,
     options: {
       logger: LoggerService;
+      events?: EventsService;
       parser?: BitbucketServerLocationParser;
       schedule?: SchedulerServiceTaskRunner;
       scheduler?: SchedulerService;
+      catalogApi?: CatalogApi;
+      tokenManager?: TokenManager;
```

Comment:
Token manager has been removed, the new equivalent is `AuthService`. I think similar logic can be used, where if an `AuthService` is not provided then the events functionality is disabled.